src/coreComponents/linearAlgebra/interfaces/hypre/HypreSuperlu.hpp

```diff
@@ -0,0 +1,142 @@
+  int_t * rowPtr;                     //!< row pointer
+  SuperMatrix mat;                    //!< SuperLU_Dist matrix format
+  dScalePermstruct_t ScalePermstruct; //!< data structure to scale and permute the matrix
+  dLUstruct_t LUstruct;               //!< data structure to store the LU factorization
+  SuperLUStat_t stat;                 //!< data structure to gather some statistics
+  gridinfo_t grid;                    //!< SuperLU_Dist MPI subdivision of load
+  dSOLVEstruct_t SOLVEstruct;         //!< data structure to solve the matrix
+  superlu_dist_options_t options;     //!< SuperLU_Dist options
+  MPI_Comm comm;                      //!< MPI communicator
+};
+
+/**
+ * @brief Converts a matrix from Hypre to SuperLU_Dist format
+ * @param[in] matrix the HypreMatrix object
+ * @param[out] SLUDData the structure containing the matrix in SuperLU_Dist format
+ */
+void ConvertToSuperMatrix( HypreMatrix const & matrix,
+                           SuperLU_DistData & SLUDData );
+
+/**
+ * @brief Creates the SuperLU_Dist data structure
+ * @param[in] matrix the HypreMatrix object
+ * @param[in] params the linear solver parameters
+ * @param[out] SLUDData the structure containing the matrix in SuperLU_Dist format
+ */
+void SuperLU_DistCreate( HypreMatrix const & matrix,
+                         LinearSolverParameters const & params,
+                         SuperLU_DistData & SLUDData );
+
+/**
+ * @brief Factorizes a linear system with SuperLU_Dist
+ * @param[in,out] SLUDData the structure containing the matrix in SuperLU_Dist format
+ * @param[out] time time spent in the factorization phase
+ * @return info error code
+ */
+int SuperLU_DistSetup( SuperLU_DistData & SLUDData,
+                       real64 & time );
+
+/**
+ * @brief Solves a linear system with SuperLU_Dist (matrix has already been factorized)
+ * @param[in,out] SLUDData the structure containing the matrix in SuperLU_Dist format
+ * @param[in] b the right-hand side in Hypre format
+ * @param[out] x the solution in Hypre format
+ * @param[out] time time spent in the solution phase
+ * @return info error code
+ */
+int SuperLU_DistSolve( SuperLU_DistData & SLUDData,
+                       HypreVector const & b,
+                       HypreVector & x,
+                       real64 & time );
+
+/**
+ * @brief Deallocates a SuperLU_Dist data structure
+ * @param[in,out] SLUDData the structure containing the matrix in SuperLU_Dist format
+ */
+void SuperLU_DistDestroy( SuperLU_DistData & SLUDData );
+
+/**
+ * @brief Converts from GEOSX to SuperLU_Dist columns permutation option
+ * @param[in] value the GEOSX option
+ * @return the SuperLU_Dist option
+ */
+colperm_t const & HypreGetColPermType( string const & value );
+
+/**
+ * @brief Converts from GEOSX to SuperLU_Dist rows permutation option
+ * @param[in] value the GEOSX option
+ * @return the SuperLU_Dist option
+ */
+rowperm_t const & HypreGetRowPermType( string const & value );
```

Comment:
I suppose these functions are internal to `HypreSuperlu`, i.e. not called from anywhere else. If so, I'd typically not expose them in the header, and even wrap the implementations in `.cpp` file into an [unnamed namespace](https://en.cppreference.com/w/cpp/language/namespace#Unnamed_namespaces) to avoid external linkage.